artemis-protocols/artemis-stomp-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/stomp/StompSession.java

```diff
@@ -79,6 +79,11 @@
       this.consumerCredits = ConfigurationHelper.getIntProperty(TransportConstants.STOMP_CONSUMERS_CREDIT, TransportConstants.STOMP_DEFAULT_CONSUMERS_CREDIT, connection.getAcceptorUsed().getConfiguration());
    }
 
+   @Override
+   public boolean supportsDirectDelivery() {
```

Comment:
if its a deadlock issue in queue, this shouldn't be needed. We should avoid using a sledge hammer by disabling features. fix the deadlock not just disable stuff.

Reply:
Same of mqtt

Reply:
and same response.

Reply:
the issue is that the protocol is holding locks on both places (consumer and queue), hence it cannot support direct deliver.